### PR TITLE
Feature: type "function" and raw attributes

### PR DESCRIPTION
**Pull request by tiagomapmarques**

### CODE REVIEW

File: docs/component.md [hunk 1]
```diff
@@ -268,6 +271,32 @@ MyButton.register();
...
+  ...
+}
+
+...
+MyButton.attributes = [
+  {
+    name: 'primary',
+    converter: (prop) => {
+      const parsedProp = toBoolean(prop);
+      console.log('original prop:', prop, ', parsed prop:', parsedProp);
```
Comment (jurekbarth) [inline docs/component.md]: I don't know if we need the `console.log` here. Or you might add another example which is super short for _real_ usage. Like:

```
converter: toBoolean
```

File: src/attribute-converters.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+export const toFunction = (prop?: PropValue): Function => {
+  if (typeof prop === 'function') {
+    return prop;
+  }
+  try {
+    // eslint-disable-next-line no-new-func
+    return (new Function(`return ${prop};`))();
+  } catch (_) {
+    return null;
```
Comment (jurekbarth) [inline src/attribute-converters.ts]: That's a silent error. Can we throw an error in the console? And as soon as we have a development bundle and a production bundle we can change the behaviour to throw in development?

Comment (tiagomapmarques) [reply src/attribute-converters.ts]: Here I disagree. The converters should do their best to convert to that specific type. It just so happens that `new Function` sometimes throws an error. But that's a detail of the implementation itself. IMO the output of the converters should always be "the type the dev requested" or "null" - nothing else (error throwing included).

File: docs/component.md (hunk 1 above)
Comment (tiagomapmarques) [reply docs/component.md]: Fixed :)

File: src/attribute-converters.ts (hunk 2 above)
Comment (jurekbarth) [reply src/attribute-converters.ts]: Fair point, let's see how it goes and add a warning to the docs if needed.

**Pull request merged**